crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+        match RwLockReadGuard::try_map(guard, |v| v.as_ref()) {
+            Ok(v) => v,
+            Err(_) => unreachable!("the connections cache must have been initialized"),
+        }
+    }
+}
+
+/// Configuration for connecting a nats client.
+/// More options are available if you use the json than variables in the values string map.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct ConnectionConfig {
+    /// URIs used to connect to the cluster
+    #[serde(default)]
+    cluster_uris: Vec<String>,
+
+    /// Authentication JWT
+    #[serde(default)]
+    auth_jwt: Option<String>,
+
+    /// Authentication Seed
+    #[serde(default)]
+    auth_seed: Option<String>,
+
+    /// Prefix for the lattice to use
+    #[serde(default)]
+    lattice_prefix: String,
+
+    /// NATS JetStream domain
+    #[serde(default)]
+    js_domain: Option<String>,
+
+    /// Operation timeout used for the lattice client interface
+    timeout_ms: u64,
+
+    /// Auction timeout used for the lattice client interface
+    auction_timeout_ms: u64,
+}
+
+impl Default for ConnectionConfig {
+    fn default() -> Self {
+        Self {
+            cluster_uris: vec![DEFAULT_NATS_URI.to_owned()],
+            auth_jwt: None,
+            js_domain: None,
+            auth_seed: None,
+            lattice_prefix: String::from(DEFAULT_LATTICE_PREFIX),
+            timeout_ms: DEFAULT_TIMEOUT_MS,
+            auction_timeout_ms: DEFAULT_AUCTION_TIMEOUT_MS,
+        }
+    }
+}
+
+/// Implement the basic requirements of a wasmcloud capability provider
+#[async_trait]
+impl WasmcloudCapabilityProvider for LatticeControllerProvider {
+    #[instrument(level = "debug", skip(self, _ld), fields(actor_id = %_ld.actor_id))]
+    async fn put_link(&self, _ld: &wasmcloud_provider_sdk::core::LinkDefinition) -> bool {
+        // This provider is *externally* multiplexed -- link definitions
+        // contain no useful data
+        true
+    }
+
+    /// Handle notification that a link is dropped
+    #[instrument(level = "debug", skip(self), fields(actor_id = ?_actor_id))]
+    async fn delete_link(&self, _actor_id: &str) {
+        // Link definitions do not determine the NATS connections
+        // so there is nothing to clean up per link removal
+    }
+
+    /// Handle shutdown request
+    #[instrument(level = "debug", skip(self))]
+    async fn shutdown(&self) {
+        // No cleanup necessary for shutdown, since links do not
+        // determine/regulate the NATS connections
+    }
+}
+
+/// Implement the lattice-controller-provider provider contract specified in WIT (see provider.wit)
+#[async_trait]
+impl WasmcloudLatticeControlLatticeController for LatticeControllerProvider {
+    /// Sets lattice credentials and stores them in the cache to be used to create a
+    /// connection in the next operation that requires one
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
```

Comment:
Nit: Change this to `skip_all`

Reply:
Yup will do

- [ ] use `skip_all` for `instrument`